addons/test_mail/tests/test_mail_composer.py

```diff
@@ -453,6 +453,20 @@ def test_mail_composer_rights_attachments(self):
             sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
             sorted(template_1_attachment_name))
 
+    @users('employee')
+    def test_mail_composer_template_without_sender(self):
+        """ Ensure that current user is used as sender when template is changed template with sender to the one without it"""
+        # Step 1: use template with sender
+        template_with_sender = self.template.copy({'email_from': 'not_current_user@test.com'})
+        composer = self.env['mail.compose.message'].create({'template_id': template_with_sender.id})
+        composer._onchange_template_id_wrapper() # here composer.template_id is set to mail.template(94,)
```

Comment:
Not sure why this doesn't work but you could use a `Form(composer)` to get a "front-end-like" api which is more relevant for the fixed use case anyway.

Reply:
> Not sure why this doesn't work

I'm not sure what do you mean? What isn't working in that part of the test?

PS: I would like to expand this test to check also mass_mailing mode probably

Reply:
Sorry, I meant "I don't know why onchange isn't triggered" basically. Normally you want to avoid calling it manually as it hides the issue that it's not being triggered correctly.